### PR TITLE
[SPARK-44392][SQL][TESTS] Add tests for schema pruning in delta-based UPDATEs

### PR DESCRIPTION
**Pull request by aokolnychyi**
### What changes were proposed in this pull request?

This PR adds tests for schema pruning for delta-based UPDATEs, similar to the ones we have for MERGE.

### Why are the changes needed?

These changes are needed to verify schema pruning works as expected.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

This PR comes with tests.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DeltaBasedMergeIntoTableSuiteBase.scala [hunk 1]
```diff
@@ -215,18 +212,4 @@ abstract class DeltaBasedMergeIntoTableSuiteBase extends MergeIntoTableSuiteBase
           Row(3, 300, "china", "software"))) // insert
     }
   }
-
-  private def executeAndCheckScan(
```
Comment (aokolnychyi) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DeltaBasedMergeIntoTableSuiteBase.scala:219]: Moving into the parent class to reuse for UPDATEs.

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DeltaBasedUpdateAsDeleteAndInsertTableSuite.scala [hunk 2]
```diff
@@ -17,44 +17,12 @@
...
+class DeltaBasedUpdateAsDeleteAndInsertTableSuite extends DeltaBasedUpdateTableSuiteBase {
 
   override protected lazy val extraTableProps: java.util.Map[String, String] = {
     val props = new java.util.HashMap[String, String]()
     props.put("supports-deltas", "true")
     props.put("split-updates", "true")
     props
   }
-
-  test("nullable row ID attrs") {
```
Comment (aokolnychyi) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DeltaBasedUpdateAsDeleteAndInsertTableSuite.scala:31]: Moved these tests into `DeltaBasedUpdateTableSuiteBase` to reduce duplication.

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DeltaBasedUpdateTableSuiteBase.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      errorClass = "INVALID_NON_DETERMINISTIC_EXPRESSIONS",
+      parameters = Map("sqlExprs" -> "\"((id <= 1) AND (rand() > 0.5))\""),
+      context = ExpectedContext(
+        fragment = "UPDATE cat.ns1.test_table SET dep = 'invalid' WHERE id <= 1 AND rand() > 0.5",
+        start = 0,
+        stop = 75)
+    )
+  }
+
+  test("update with schema pruning") {
```
Comment (aokolnychyi) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DeltaBasedUpdateTableSuiteBase.scala:74]: This is the new test.

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for adding more test cases.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by aokolnychyi** [comment]
Thank you, @dongjoon-hyun!

**Pull request closed without merging**